packages/devtools/src/commands/performActions.js

```diff
@@ -0,0 +1,109 @@
+import USKeyboardLayout from 'puppeteer-core/lib/USKeyboardLayout'
+
+const sleep = (time = 0) => new Promise(
+    (resolve) => setTimeout(resolve, time))
+
+export default async function performActions ({ actions }) {
+    const page = this.getPageHandle()
+    const lastPointer = {}
+
+    /**
+     * see https://github.com/jlipps/simple-wd-spec#input-sources-and-corresponding-actions
+     * for details on the `actions` format
+     */
+    for (const action of actions) {
+        if (action.type === null || action.type === 'null') {
+            for (const singleAction of action.actions) {
+                await sleep(singleAction.duration)
+            }
+            continue
+        }
+
+        if (action.type === 'key') {
+            const skipChars = []
+            for (const singleAction of action.actions) {
+                if (singleAction.type === 'pause') {
+                    await sleep(singleAction.duration)
+                    continue
+                }
+
+                const cmd = singleAction.type.slice(3).toLowerCase()
+                const keyboardFn = ::page.keyboard[cmd]
```

Comment:
should we check if cmd is one of type/press/down/up/sendCharacter ?

ref https://github.com/puppeteer/puppeteer/blob/v2.1.1/docs/api.md#class-keyboard

Reply:
I use either `down`/`up` or `sendCharacter` as I thought they would best represent what the protocol is requesting.